src/resmom/requests.c

```diff
@@ -619,23 +620,27 @@ req_deletejob(struct batch_request *preq)
 	 * mom_deljob_wait() sets substate to
 	 * prevent sending more OBIT messages
 	 */
+	pjob->ji_preq = preq;
+	hook_input = (mom_hook_input_t *) malloc (sizeof(mom_hook_input_t));
+	if (hook_input == NULL) {
+		log_err(errno, __func__, MALLOC_ERR_MSG);
+		return;
+	}
+	mom_hook_input_init(hook_input);
+	hook_input->pjob = pjob;
 
-#if MOM_ALPS
-	pjob->ji_preq = preq;	/* needed when canceling ALPS */
-#else
-	pjob->ji_preq = NULL;
-#endif
-	mom_hook_input_init(&hook_input);
-	hook_input.pjob = pjob;
-
-	mom_hook_output_init(&hook_output);
-	hook_output.reject_errcode = &reject_errcode;
-	hook_output.last_phook = &last_phook;
-	hook_output.fail_action = &hook_fail_action;
+	if (mom_process_hooks(HOOK_EVENT_EXECJOB_END,
+		PBS_MOM_SERVICE_NAME, mom_host, hook_input,
+		hook_output, hook_msg, sizeof(hook_msg), 1) == HOOK_RUNNING_IN_BACKGROUND) {
+			/* 
+			 * Hook is running in background reply to the batch
+			 * request will be taken care of in run_execjob_end_hooks
+			 * function 
+			 */
+			pjob->ji_execjob_end_hook_event_started = 1;
+			return;
```

Comment:
Does this mean that the MS will run her hooks first, then only when she's done tell her sisters to run the hooks?

Reply:
Yes and I see no difference in sequence after the code change.

Reply:
As far as I know, the reason why mother superior ran her hooks first, then the sisters, was because the hook would block the mom, preventing her from sending the IM_DELETE_JOB_REPLY requests to her sisters. Now that the hook doesn't block, shouldn't the request be sent earlier, before the hook finishes?

Reply:
Ah, I overlooked this, thanks for pointing it out, I will address this in the next commit.